invenio_jobs/services/services.py

```diff
@@ -266,11 +284,134 @@ def create(self, identity, job_id, data, uow=None):
+    def add_total_entries(self, identity, run_id, job_id, total_entries, uow=None):
+        """Increment the total entries of a run."""
+        self.require_permission(identity, "update")
+        run = get_run(run_id=run_id, job_id=job_id)
+
+        if not isinstance(total_entries, int):
+            raise ValueError("total_entries must be an integer")
+
+        if total_entries < 0:
+            raise ValueError("total_entries cannot be negative")
+
+        run.total_entries += total_entries
+        uow.register(ModelCommitOp(run))
+        return self.result_item(self, identity, run, links_tpl=self.links_item_tpl)
+
+    @unit_of_work()
+    def create_subtask_run(self, identity, parent_run_id, job_id, args=None, uow=None):
+        """Create a new subtask Run."""
+        self.require_permission(identity, "create")
+        parent_run = get_run(run_id=parent_run_id, job_id=job_id)
+        job = parent_run.job
+
+        subtask_run = Run.create(
+            job=job,
+            id=str(uuid.uuid4()),
+            task_id=str(uuid.uuid4()),
+            started_by_id=identity.id,
+            status=RunStatusEnum.QUEUED,
+            title=f"Run {parent_run_id} — Subtask",
+            args=args or {},
+        )
+
+        parent_run.total_subtasks += 1
+        subtask_run.parent_run_id = parent_run.id
+        uow.register(ModelCommitOp(subtask_run))
+        uow.register(ModelCommitOp(parent_run))
+        return self.result_item(
+            self, identity, subtask_run, links_tpl=self.links_item_tpl
+        )
+
+    @unit_of_work()
+    def start_processing_subtask(self, identity, run_id, job_id, uow=None):
+        """Start processing a subtask."""
+        self.require_permission(identity, "update")
+        run = get_run(run_id=run_id, job_id=job_id)
+
+        if run.status != RunStatusEnum.QUEUED:
+            raise RunStatusChangeError(run, RunStatusEnum.RUNNING)
+
+        run.status = RunStatusEnum.RUNNING
+        run.started_at = db.func.now()
+
+        uow.register(ModelCommitOp(run))
+        return self.result_item(self, identity, run, links_tpl=self.links_item_tpl)
+
+    @unit_of_work()
+    def finalize_subtask(
+        self,
+        identity,
+        run_id,
+        job_id,
+        errored_entries_count=0,
+        success=True,
+        message=None,
+        uow=None,
+    ):
+        """Finalize a subtask and update its parent."""
+        self.require_permission(identity, "update")
+        run = get_run(run_id=run_id, job_id=job_id)
+        # Lock parent row to prevent concurrent updates
+        parent = (
+            db.session.query(Run)
+            .filter(Run.id == run.parent_run_id)
+            .with_for_update()
+            .first()
+        )
+        run.status = RunStatusEnum.SUCCESS if success else RunStatusEnum.FAILED
+
+        parent.errored_entries += errored_entries_count
+        parent.completed_subtasks += 1
+        if not success:
+            parent.failed_subtasks += 1
```

Comment:
minor: this is affected by the same issue as the above method has in terms of updating the error and completed count columns